nimbus/src/dbcache.rs

```diff
@@ -84,7 +92,10 @@ impl DatabaseCache {
         }
     }
 
-    pub fn get_experiment_branch(&self, slug: &str) -> Result<Option<String>> {
-        self.get_data(|data| data.experiment_branches.get(slug).cloned())
+    pub fn get_experiment_branch(&self, id: &str) -> Result<Option<String>> {
+        self.get_data(|data| match data.branches_by_feature.get(id) {
+            None => data.branches_by_experiment.get(id).cloned(),
+            Some(branch_slug) => Some(branch_slug.to_owned()),
+        })
```

Comment:
😻

Reply:
Thanks to @rfk for helping me sort this out!